CDP4CrossViewEditor/Assemblers/CrossviewArrayAssembler.cs

```diff
@@ -340,17 +344,70 @@ private List<object[]> GenerateContentHeader()
+
+                            if (isCalculationPossible &&
+                                CrossviewSheetConstants.PowerParameters.Contains(parameterOrOverrideBase.ParameterType.ShortName))
+                            {
+                                calculationParameters.Add((parameterValueSetBase, component));
+                            }
                         }
                     }
                     else
                     {
                         var index = this.GetContentColumnIndex(parameterValueSetBase);
                         contentRow[index] = parameterValueSetBase.ActualValue.First();
                         namesRow[index] = $"{CrossviewSheetConstants.CrossviewSheetName}_{parameterValueSetBase.ModelCode()}";
+
+                        if (!isCalculationPossible)
+                        {
+                            continue;
+                        }
+
+                        if (CrossviewSheetConstants.PowerParameters.Contains(parameterOrOverrideBase.ParameterType.ShortName))
+                        {
+                            calculationParameters.Add((parameterValueSetBase, null));
+                        }
                     }
                 }
             }
 
+            if (!isCalculationPossible)
+            {
+                return (contentRow, namesRow);
+            }
+
+            var indexes = CrossviewSheetConstants.PowerParameters.Select(p => this.GetContentColumnsIndexes(calculationParameters, p)).SelectMany(x => x).ToList();
+
+            var rSchemeIndex = indexes.FirstOrDefault(p => p.Item1.Equals("redundancy.scheme", StringComparison.InvariantCultureIgnoreCase)).Item2;
+            var rTypeIndex = indexes.FirstOrDefault(p => p.Item1.Equals("redundancy.type", StringComparison.InvariantCultureIgnoreCase)).Item2;
+            var rKIndex = indexes.FirstOrDefault(p => p.Item1.Equals("redundancy.k", StringComparison.InvariantCultureIgnoreCase)).Item2;
+            var rNIndex = indexes.FirstOrDefault(p => p.Item1.Equals("redundancy.n", StringComparison.InvariantCultureIgnoreCase)).Item2;
+
+            var pOnIndex = indexes.FirstOrDefault(p => p.Item1.Equals("P_on", StringComparison.InvariantCultureIgnoreCase)).Item2;
+            var pStandbyIndex = indexes.FirstOrDefault(p => p.Item1.Equals("P_stby", StringComparison.InvariantCultureIgnoreCase)).Item2;
+
+            var pDutyCycleIndexes = indexes.Where(p => p.Item1.Equals("P_duty_cyc", StringComparison.InvariantCultureIgnoreCase)).ToArray();
+            var pMeanIndexes = indexes.Where(p => p.Item1.Equals("P_mean", StringComparison.InvariantCultureIgnoreCase)).ToArray();
+
+            if (pMeanIndexes.Length != pDutyCycleIndexes.Length)
+            {
+                // P_duty_cyc & P_mean must have the same state dependancies and the same options
+                return (contentRow, namesRow);
+            }
+
+            for (var i = 0; i < pDutyCycleIndexes.Count(); i++)
```

Comment:
This assumes that for pMean and pDutyCycle values each index corresponds to the same option/state. Which could technically be false, as there's no guarantee about the order of these.